ideas/single_connection_usage_with_timeouts.cpp

```diff
@@ -0,0 +1,15 @@
+boost::asio::io_service io_service;
+
+boost::asio::spawn(io_service, [&] (boost::asio::yield_context yield) {
+    auto connection = apq::make_connection_provider(io_service);
+    apq::cursor<apq::row> cursor;
+    connection_timeouts timeouts;
+    timeouts.connect = 100ms;
+    timeouts.request = 200ms;
+    apq::text_protocol::request(connection, apq::make_query("SELECT pg_is_in_recovery()", timeouts), cursor, yield);
```

Comment:
About timeouts. In this case we allow to specify custom timeouts for the query. It is useful future and IMO it is better to provided rather than force a user to make his (or hers) own solution.